### PR TITLE
[SILInliner] Set the correct scope for {begin,end}_borrow.

### PR DESCRIPTION
**Pull request by dcci**
<rdar://problem/51512044>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test

**Comment by ravikandhadai** [comment]
Looks great to me! Thanks.

**Review by ravikandhadai — APPROVED** [review]

**Review by gottesmm — APPROVED** [review]
Looks great! Thanks!

File: lib/SILOptimizer/Utils/SILInliner.cpp [hunk 1]
```diff
@@ -559,15 +559,18 @@ SILValue SILInlineCloner::borrowFunctionArgument(SILValue callArg,
       || callArg.getOwnershipKind() != ValueOwnershipKind::Owned) {
     return callArg;
   }
-  auto *borrow = getBuilder().createBeginBorrow(AI.getLoc(), callArg);
+
+  SILBuilderWithScope beginBuilder(AI.getInstruction());
```
Comment (gottesmm) [inline lib/SILOptimizer/Utils/SILInliner.cpp]: I was just thinking about this. Should we pass down the SILBuilderContext here?

Comment (dcci) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: My take is that creating a `SILBuilderWithScope` is _extremely_ cheap so, given we don't really know whether the instructions down will have the same scope of the `apply`, we might as well create a fresh object.

Comment (gottesmm) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: The reason I am asking is there might be code that is passing in an InsertedInstrs
or something like that.

**Review by ravikandhadai — APPROVED** [review]

File: lib/SILOptimizer/Utils/SILInliner.cpp (hunk 1 above)
Comment (dcci) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: I might be wrong here so please correct me, but I think this doesn't really matter here. We create new `SILBuilderWithScopes` on instructions that aren't inserted.

**Comment by dcci** [comment]
@swift-ci test

**Comment by dcci** [comment]
@swift-ci test and merge

File: lib/SILOptimizer/Utils/SILInliner.cpp (hunk 1 above)
Comment (dcci) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: Resolved this in person & changed the code to pass a `SILBuilderContext` around.

**Comment by dcci** [comment]
@swift-ci test and merge

File: lib/SILOptimizer/Utils/SILInliner.cpp [hunk 2]
```diff
@@ -559,15 +559,18 @@ SILValue SILInlineCloner::borrowFunctionArgument(SILValue callArg,
       || callArg.getOwnershipKind() != ValueOwnershipKind::Owned) {
     return callArg;
   }
-  auto *borrow = getBuilder().createBeginBorrow(AI.getLoc(), callArg);
+
+  SILBuilderWithScope beginBuilder(AI.getInstruction(), getBuilder());
+  auto *borrow = beginBuilder.createBeginBorrow(AI.getLoc(), callArg);
   if (auto *tryAI = dyn_cast<TryApplyInst>(AI)) {
-    SILBuilder returnBuilder(tryAI->getNormalBB()->begin());
+    SILBuilderWithScope returnBuilder(tryAI->getNormalBB()->begin());
```
Comment (gottesmm) [inline lib/SILOptimizer/Utils/SILInliner.cpp]: Can you pass in the builder to /all/ of these so they all use the same builder context?

Comment (dcci) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: Wouldn't that change the semantic though? The other builders didn't do that before.

**Comment by dcci** [comment]
@swift-ci smoke test

**Comment by dcci** [comment]
@swift-ci smoke test

File: lib/SILOptimizer/Utils/SILInliner.cpp (hunk 2 above)
Comment (gottesmm) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: That would. And I am saying as the person who wrote this code that it was probably wrong.

Comment (dcci) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: Missed your comment. Doing this as a follow-up.

**Pull request merged**